compiler/src/main/scala/org/scalajs/nscplugin/JSGlobalAddons.scala

```diff
@@ -209,48 +211,50 @@ trait JSGlobalAddons extends JSDefinitions
     }
 
     object JSUnaryOpMethodName {
-      private val map = Map[Name, js.JSUnaryOp.Code](
-        nme.UNARY_+ -> js.JSUnaryOp.+,
-        nme.UNARY_- -> js.JSUnaryOp.-,
-        nme.UNARY_~ -> js.JSUnaryOp.~,
-        nme.UNARY_! -> js.JSUnaryOp.!
+      private val map = Map[Name, (js.JSUnaryOp.Code, Boolean)](
+        nme.UNARY_+ -> (js.JSUnaryOp.+, true),
+        nme.UNARY_- -> (js.JSUnaryOp.-, true),
+        nme.UNARY_~ -> (js.JSUnaryOp.~, true),
+        nme.UNARY_! -> (js.JSUnaryOp.!, true)
       )
 
       /* We use Name instead of TermName to work around
        * https://github.com/scala/bug/issues/11534
        */
-      def unapply(name: Name): Option[js.JSUnaryOp.Code] =
+      def unapply(name: Name): Option[(js.JSUnaryOp.Code, Boolean)] =
         map.get(name)
     }
 
     object JSBinaryOpMethodName {
-      private val map = Map[Name, js.JSBinaryOp.Code](
-        nme.ADD -> js.JSBinaryOp.+,
-        nme.SUB -> js.JSBinaryOp.-,
-        nme.MUL -> js.JSBinaryOp.*,
-        nme.DIV -> js.JSBinaryOp./,
-        nme.MOD -> js.JSBinaryOp.%,
-
-        nme.LSL -> js.JSBinaryOp.<<,
-        nme.ASR -> js.JSBinaryOp.>>,
-        nme.LSR -> js.JSBinaryOp.>>>,
-        nme.OR  -> js.JSBinaryOp.|,
-        nme.AND -> js.JSBinaryOp.&,
-        nme.XOR -> js.JSBinaryOp.^,
-
-        nme.LT -> js.JSBinaryOp.<,
-        nme.LE -> js.JSBinaryOp.<=,
-        nme.GT -> js.JSBinaryOp.>,
-        nme.GE -> js.JSBinaryOp.>=,
-
-        nme.ZAND -> js.JSBinaryOp.&&,
-        nme.ZOR  -> js.JSBinaryOp.||
+      private val map = Map[Name, (js.JSBinaryOp.Code, Boolean)](
+        nme.ADD -> (js.JSBinaryOp.+, true),
+        nme.SUB -> (js.JSBinaryOp.-, true),
+        nme.MUL -> (js.JSBinaryOp.*, true),
+        nme.DIV -> (js.JSBinaryOp./, true),
+        nme.MOD -> (js.JSBinaryOp.%, true),
+
+        nme.LSL -> (js.JSBinaryOp.<<, true),
+        nme.ASR -> (js.JSBinaryOp.>>, true),
+        nme.LSR -> (js.JSBinaryOp.>>>, true),
+        nme.OR  -> (js.JSBinaryOp.|, true),
+        nme.AND -> (js.JSBinaryOp.&, true),
+        nme.XOR -> (js.JSBinaryOp.^, true),
+
+        nme.LT -> (js.JSBinaryOp.<, true),
+        nme.LE -> (js.JSBinaryOp.<=, true),
+        nme.GT -> (js.JSBinaryOp.>, true),
+        nme.GE -> (js.JSBinaryOp.>=, true),
+
+        nme.ZAND -> (js.JSBinaryOp.&&, true),
+        nme.ZOR  -> (js.JSBinaryOp.||, true),
+
+        nme.MUL.append(nme.MUL) -> (js.JSBinaryOp.**, false)
```

Comment:
Feels a bit odd to use a constant for `*` where the sole purpose of the constant is to avoid too much allocations and then create a new object with it :-/

Maybe just creating a new name here, or putting it in JSDefinitions is neater?

Reply:
It was more for the encoding than for the reusing the object. But I changed it to an explicit call to `global.encode` instead.